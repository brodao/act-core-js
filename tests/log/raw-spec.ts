import { ACT } from '../../lib/index';

const { wrap } = require('@brodao/act-jest-snapshot-console');

describe('Console em texto puro', () => {
	test("Apresenta o 'splash'", () => {
		expect(
			wrap(() =>
				ACT.logger.showHeader({
					name: 'test_show_banner',
					version: '99.99.99',
					description: 'Show Banner',
					url: 'http:://no.url',
				})
			)
		).toMatchSnapshot();
	});

	test('Aviso simples', () => {
		expect(wrap(() => ACT.logger.warn('Aviso simples'))).toMatchSnapshot();
	});

	test('Aviso com vários agumentos', () => {
		expect(
			wrap(() =>
				ACT.logger.warn('Aviso com argumentos', 'arg 1', 'arg 2', 'arg 3')
			)
		).toMatchSnapshot();
	});

	test('Erro simples', () => {
		expect(wrap(() => ACT.logger.error('Erro simples'))).toMatchSnapshot();
	});

	test('Erro com vários agumentos', () => {
		expect(
			wrap(() =>
				ACT.logger.error('Erro com argumentos', 'arg 1', 'arg 2', 'arg 3')
			)
		).toMatchSnapshot();
	});
});
