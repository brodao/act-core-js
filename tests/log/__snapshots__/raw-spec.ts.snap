// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console em texto puro Apresenta o 'splash' 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "E:[winston] Unknown logger level: %s
E:[winston] Unknown logger level: %s
E:[winston] Unknown logger level: %s
E:[winston] Unknown logger level: %s
E:[winston] Unknown logger level: %s
E:[winston] Unknown logger level: %s
E:[winston] Unknown logger level: %s
E:[winston] Unknown logger level: %s
",
}
`;

exports[`Console em texto puro Aviso com vários agumentos 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "",
}
`;

exports[`Console em texto puro Aviso simples 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "",
}
`;

exports[`Console em texto puro Erro com vários agumentos 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "",
}
`;

exports[`Console em texto puro Erro simples 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "",
}
`;
